Sample/SampleDelegate/SampleDelegate/AppDelegate.swift

```diff
@@ -33,65 +33,65 @@ class AppDelegate: SuperDelegate, ApplicationLaunched {
         // Setup app model, networking, logging, etc.
     }
     
-    func loadInterfaceWithLaunchItem(launchItem: LaunchItem) {
+    func loadInterface(launchItem: LaunchItem) {
         // Setup our main window since we don't have a storyboard.
-        setupMainWindow(window)
+        setup(mainWindow: window)
         window.rootViewController = ViewController()
         
         switch launchItem {
-        case let .RemoteNotificationItem(remoteNotification):
+        case let .remoteNotification(item):
             // Show UI for the remote notification that brought us into the foreground.
-            didReceiveRemoteNotification(remoteNotification, notificationOrigin: .UserTappedToBringAppToForeground, fetchCompletionHandler: { (_) in
+            didReceive(remoteNotification: item, origin: .userTappedToBringAppToForeground, fetchCompletionHandler: { (_) in
```

Comment:
Interesting. This raises an interesting debate around Optional completion handlers to avoid writing something like { (_, _, _) in } when they expect parameters.